buildtest/utils/file.py

```diff
@@ -143,3 +143,80 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath,str):
+        return None
+
+    # resolve_path will handle shell and user expansion and account for any symlinks and check for file existence.
+    # if resolve_path does not return gracefully it implies file does not exist and will return None
+    filepath = resolve_path(filepath)
+
+    # if it's invalid file let's return None
+    if not filepath:
+        return None
+
+    content = []
+    fd = open(filepath, "r")
```

Comment:
This is overly complicated - we just need:
```
with open(filepath, 'r') as fd:
    content = fd.read()
```
This parsing of lines doesn't make sense. If we wanted to read lines we could do readline() but it's better to return the raw content of the file and let the parsing function decide what to do. For example, if we used this and piped into a function to parse yaml, it would be all kinds of weird to get a list of strings.

Reply:
ahhh, i was kinda following the principles of read and write by line similar to ``BuildTestCommand`` . If you wanted a simple as that, i could have saved some time :(
Would you like the return type for ``read_file`` be a string or a list.

Reply:
Sorry for the misunderstanding! I literally meant changing those 2 lines into a function since we re-use, and adding a check (for read) that the file exists first, and (for write) that the directory to write exists first.

Generally, we don't expect special / custom parsing, so you'd want to use read() and just return raw content, same idea for write.

Reply:
ok if i understand i will create two methods ``read`` and ``write`` with the suggestion you proposed.